react/week2/my_first_app/src/index.js

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
+//import App from './App';
+import * as serviceWorker from './serviceWorker';
+
+const root = document.getElementById('root');
+
+const todosArray = [
+  {
+    "id": 1,
+    "description": "Get out of bed"
+  },
+  {
+    "id": 2,
+    "description": "Brush teeth"
+  },
+  {
+    "id": 3,
+    "description": "Eat breakfast"
+  }, 
+  {
+    "id": 4,
+    "description": "Understand State and props"
+  },
+  {
+    "id": 5,
+    "description": "Passing the state through props"
+  }
+];
+class Header extends React.Component {
+  render() {
+    return <h1>Todolist</h1>
+  }
+}
+//I cannot pass a props that is in a state in one component and
+//set that props again in state in different component
+class TodoList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isChecked: false,
+      checked: []
+    };
+  }
+
+  handleCheckboxChange = (event) => {
+    const temp = event.target.checked
```

Comment:
Have you considered using a map to track the checked state instead of an array? Would make this code much simpler if you had something like this in your state:
```
{ 
   checkedTodosById: {
     1: true,
     2: false
   }
}
```

and then when you're rendering your list is much simpler to know if a given todo is checked or not.

I would also add that since you're making this component controlled (as we discussed on slack) it is better if you make it with less state as possible. So the checked state of the todos is also controlled by the parent component. That's a plus though. Not required for this homework.

Reply:
Thanks for your suggestion. I will try to think this idea.